docs/proposals/algorithms/joint-inference/Proposal-Toward-Embodied-AI-Applications.md

```diff
@@ -0,0 +1,329 @@
+
+#### **Quick Action Reasoning Module**
+The edge model is augmented with a lightweight reasoning head that translates segmentation results into immediate action commands, ensuring sub-50ms response times for critical safety scenarios:
+
+**Safety-Critical Action Rules:**
+- **STOP**: Triggered when pedestrians are detected within 5-meter proximity or unexpected obstacles exceed confidence threshold (>0.8)
+- **SLOW**: Activated for vehicles, cyclists, or construction zones within the path corridor
+- **NAVIGATE**: Issued for clear path scenarios with standard road markings
+- **ALERT**: Generated for ambiguous objects requiring cloud-based reasoning
+
+**Implementation:** A rule-based decision tree processes segmentation masks and depth information to generate action commands within 15ms, operating in parallel with cloud communication. For example, detecting a pedestrian mask with depth < 5m immediately triggers a "STOP" command while simultaneously sending scene data to the cloud for higher-level path replanning.
+
+This dual-layer approach ensures that critical safety actions are never delayed by network latency while maintaining the system's ability to perform complex reasoning through cloud collaboration.
+### 2. Cloud Model — **Large Vision-Language Models** (GPT-4o & Qwen-VL)
+|  | GPT-4o-mini (Vision) | Qwen-VL-Max |
+|---|---------------------|--------------|
+| **Input** | RGB image & ≤ 1 Mpx video frames; text ≤ 32 k tokens | RGB < 2 Mpx; text ≤ 16 k tokens |
+| **Multimodal skills** | Spatial reasoning, chart reading, code explanation, conversation memory | CN/EN VQA, math reasoning, long-image OCR |
+| **License** | Closed API (OpenAI) | Apache-2.0, self-hostable |
+| **Why chosen** | SOTA all-round ability, multi-lingual | Open source, 24 GB GPU friendly |
+#### 
+### 3. Cloud-Edge Intelligent Decision 
+
+The Cloud-Edge Intelligent Decision module serves as a critical optimization layer between edge perception and cloud reasoning, dynamically determining the optimal resolution for uploading segmentation masks to conserve bandwidth and improve system efficiency. After RGB images are processed by the edge RFNet model to generate segmentation masks, this module intelligently classifies each mask into three resolution tiers: **low** (640×640), **medium** (1280×760), and **high** (2048×1024) resolution for cloud transmission.
+
+#### 3.1 Architecture Overview
+
+The decision module operates on the principle that not all scenes require high-resolution analysis. Simple scenes with minimal objects can be adequately processed at lower resolutions, while complex scenes containing critical safety elements demand full-resolution analysis. This adaptive approach achieves up to **70% bandwidth reduction** while maintaining decision quality.
+
+![alt text](images/image-20250812.png)
```

Comment:
Need to highlight the main modified part. We see the main modification would be the hard example mining and cloud model, instead of edge model.